chain/network/src/view_client_adapter.rs

```diff
@@ -0,0 +1,79 @@
+use futures::future::BoxFuture;
+use futures::Future;
+use near_network_primitives::types::{NetworkViewClientMessages, NetworkViewClientResponses};
+use std::fmt;
+use std::sync::Arc;
+
+/// [`ViewClientAdapter`] defines the subset of view client's interface which is
+/// needed by the network.
+///
+/// It exists primarily to break inter-crate dependencies -- network should not
+/// know the specific type of the client.
+///
+/// [`ViewClientAdapter`] is a [`Clone`] handle to some concurrent object
+/// running in a different thread or task, so [`ViewClientAdapter::send`] is
+/// `async`.
+///
+/// In theory, sending can fail if the other side has already shut down. At the
+/// moment, we don't impose any lifetimes constraints here, and generally just
+/// ignore this case.
+///
+/// Implementation wise, this is a struct which holds a `dyn Trait` inside. As
+/// we only need to customize a single method (sending a message), a
+/// [`ViewClientAdapter`] can be constructed from arbitrary closure.
+#[derive(Clone)]
+pub struct ViewClientAdapter {
+    inner: Arc<
+        dyn Fn(
+                NetworkViewClientMessages,
+            )
+                -> BoxFuture<'static, Result<NetworkViewClientResponses, ViewClientIsDeadError>>
+            + Send
+            + Sync,
```

Comment:
I guess here 'static is the default, but can you write it down explicitly?

Reply:
done